### PR TITLE
MM-54356: In-correct User count when adding members that already exists to the channel

### PR DESCRIPTION
**Pull request by alexdecamillo**
#### Summary

This PR prevents the redux actions that occur after a user addition from executing if the user is already in the channel

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/24528
Jira https://mattermost.atlassian.net/browse/MM-54356

#### Screenshots

mattermost-user-channel

#### Release Note

```release-note
Fixed a bug where the channel member count increased when adding an already present user
```

### CODE REVIEW

File: webapp/channels/src/packages/mattermost-redux/src/actions/channels.ts [hunk 1]
```diff
@@ -1049,20 +1049,23 @@ export function addChannelMember(channelId: string, userId: string, postRootId =
...
-                type: ChannelTypes.RECEIVED_CHANNEL_MEMBER,
-                data: member,
-            },
-            {
-                type: ChannelTypes.ADD_CHANNEL_MEMBER_SUCCESS,
-                id: channelId,
-            },
-        ], 'ADD_CHANNEL_MEMBER.BATCH'));
+        const membersInChannel = getState().entities.channels.membersInChannel[channelId];
+        if (!membersInChannel || !(userId in membersInChannel)) {
```
Comment (devinbinnie) [inline webapp/channels/src/packages/mattermost-redux/src/actions/channels.ts:1053]: We can simplify a bit here:

```suggestion
        if (!(membersInChannel && userId in membersInChannel)) {
```

**Review by devinbinnie — APPROVED** [review]
LGTM! Just one non-blocking comment.
Thanks @alexdecamillo!

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]
Thank you pending Devin's comment

**Review by M-ZubairAhmed — APPROVED** [review]
This looks good :shipit: Thank you

**Review by yasserfaraazkhan — APPROVED** [review]
thank you for the changes! working as expected!

**Pull request merged**